pkg/testframework/console.go

```diff
@@ -0,0 +1,162 @@
+package testframework
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	// operatorapi "github.com/openshift/api/operator/v1"
+	operatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+	v1alpha1 "github.com/openshift/console-operator/pkg/apis/console/v1alpha1"
+)
+
+func isOperatorManaged(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Managed
+}
+
+func isOperatorUnmanaged(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Unmanaged
+}
+
+func isOperatorRemoved(cr *v1alpha1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1alpha1.Removed
+}
+
+type operatorStateReactionFn func(cr *v1alpha1.Console) bool
+
+func ensureConsoleIsInDesiredState(logger Logger, client *Clientset, state operatorsv1alpha1.ManagementState) error {
+	var cr *v1alpha1.Console
+	// var checkFunc func()
+	var checkFunc operatorStateReactionFn
+
+	switch state {
+	case operatorsv1alpha1.Managed:
+		checkFunc = isOperatorManaged
+	case operatorsv1alpha1.Unmanaged:
+		checkFunc = isOperatorUnmanaged
+	case operatorsv1alpha1.Removed:
+		checkFunc = isOperatorRemoved
+	}
+
+	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		cr, err = client.ConsoleV1alpha1Interface.Consoles(consoleapi.OpenShiftConsoleOperatorNamespace).Get(consoleapi.ResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return checkFunc(cr), nil
+	})
+	if err != nil {
+		DumpObject(logger, "the latest observed state of the console resource", cr)
+		DumpOperatorLogs(logger, client)
+		return fmt.Errorf("failed to wait to change console operator state to 'Removed': %s", err)
```

Comment:
This branch is always `operator state to 'Removed'`?

Reply:
nope, will update ! :)